sdks/java/core/src/main/java/org/apache/beam/sdk/coders/AvroCoder.java

```diff
@@ -114,7 +114,7 @@
    * @param <T> the element type
    */
   public static <T> AvroCoder<T> of(Class<T> clazz) {
-    return new AvroCoder<>(clazz, ReflectData.get().getSchema(clazz));
```

Comment:
Yes, I realized this, because even though `getSchema` does touch the problematic caches, it will just result in a cache miss, I think. I would like to leave the change just to treat `ReflectData.get()` as forbidden.